src/etl/ETLState.h

```diff
@@ -36,9 +36,22 @@ struct ETLState {
 
     /**
      * @brief Fetch the ETL state from the rippled server
+     * @param source The source to fetch the state from
+     * @return The ETL state, nullopt if source not available
      */
-    static ETLState
-    fetchETLStateFromSource(Source const& source) noexcept;
+    template <typename Forward>
```

Comment:
Why did this method become template?

Reply:
This function will be called by the single source at the beginning and by LoadBalancer at retry.